### PR TITLE
feat: [CDS-56525]: Support for configFile builtin variable when more files attached to config file

### PR DESCRIPTION
**Pull request by ivan-83**
Add support for configFile built-in variable to get the content of files from Harness File Store, Encrypted Files, and Git.

Harness File Store:
```
<+configFile.getAsString("configFileId:account:/folder1/folder2/configFile")>
<+configFile.getAsString("configFileId:org:/folder1/folder2/configFile")>
<+configFile.getAsString("configFileId:/folder1/folder2/configFile")>
```

Git:
```
<+configFile.getAsString("configFileId:ssh-winrm/configSshFile.yml")>
<+configFile.getAsString("configFileId:ssh-winrm/configSshFile2.yml")>
```

Encrypted Secret Files:
```
<+configFile.getAsString("configFileId:account.AzureFileSecret")>
<+configFile.getAsString("configFileId:org.AzureFileSecret")>
<+configFile.getAsString("configFileId:AzureFileSecret")>
```
 
If the user provides a config file id as a reference like this
`<+configFile.getAsString("configFileId")>`

In this case, the file content will be returned if only one file is attached to the config file. If more files are attached to the config file null will be returned. 
If the user provides a reference as a folder path the result of built-in variable will be null.
`<+configFile.getAsString("configFileId:account:/folder1/folder2/configFile")> `

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46467)

### CODE REVIEW

**Comment by ivan-83** [comment]
trigger ss

**Comment by ivan-83** [comment]
trigger messagecheck

**Comment by ivan-83** [comment]
trigger messagecheck

**Comment by ivan-83** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/pms/expressions/functors/ConfigFileFunctor.java [hunk 1]
```diff
@@ -176,4 +203,49 @@ private String getSecretFileContentAsString(Ambiance ambiance, final String ref)
...
+      throw new InvalidArgumentsException(
+          format("Found more git files attached to config file, configFileIdentifier: %s", configFileIdentifier));
+    }
+  }
+
+  private String getGitFileContentOrThrow(String configFileIdentifier, String reference, List<ConfigGitFile> gitFiles) {
+    return isEmpty(reference)
+        ? gitFiles.get(0).getFileContent()
+        : gitFiles.stream()
+              .filter(configGitFile -> configGitFile != null && reference.equals(configGitFile.getFilePath()))
```
Comment (acasian-harness) [inline 120-ng-manager/src/main/java/io/harness/pms/expressions/functors/ConfigFileFunctor.java:231]: question here: 
if the git file is under `dir/dir2/dir3/file.txt` - what would be the `configFile.getFilePath()` and what is the reference the customer needs to specify?

**Comment by acasian-harness** [comment]
trigger review

**Review by acasian-harness — APPROVED** [review]

**Comment by acasian-harness** [comment]
trigger review

**Pull request merged**